src/mrack/providers/aws.py

```diff
@@ -265,8 +303,21 @@ async def create_server(self, req):
             ],
             "TagSpecifications": [{"ResourceType": "instance", "Tags": taglist}],
         }
-        if specs.get("subnet_id"):
-            request["SubnetId"] = specs.get("subnet_id")
+
+        if subnet_ids := specs.get("subnet_ids"):
+            for subnet_id in subnet_ids:
+                available_ips = self.get_subnet_available_ips(subnet_id, log_msg_start)
```

Comment:
Atm the biggest problem of AWS provider (i.e. not this PR) is that it is not truly asynchronous as e.g. OpenStack provider. It would be good to get to the async nature (but I'm not pushing for it here). E.g. in the method which should be asynchronous we are calling 2 blocking synchronous calls - getting the IPs and and then provisioning the instance.

I'd rather skip the getting of IPs here. It will probably not be reliable anyway, thus should be OK to do once at the beginning of provisioning, e.g. in prepare_provisioning then use it in can_provision. We could use similar methods in distribution as in openstack. But I think it is good enough to get all IPs from a single network and fail if there isn't any available.  The limit can be re-checked/updated on failure of provisioning.

Reply:
I'm OK with this async approach, but I think it would incompatible with https://github.com/neoave/mrack/pull/201#discussion_r1005968388 
So we would have to pick.

Reply:
I left a comment there for you @dav-pascual lets continue to work that both are possible :muscle:

Reply:
I pushed new changes addressing the comments.
I'm also taking a look at the async way.

Reply:
I deleted the sync calls to AWS API from create_server. 
I also made the loop calling to AWS API in `can_provision` async with gather function.
Please let me know if anything can be improved further to make it async.

I have observed that we could make the loop in `_provision_base` func async similarly as I have done, although that would be better for another PR.

Reply:
Fine for now I would say